avocado/utils/asset.py

```diff
@@ -62,7 +62,7 @@ def __init__(self, name, asset_hash=None, algorithm=None, locations=None,
         :param expire: time in seconds for the asset to expire
         :param metadata: metadata which will be saved inside metadata file
         """
-        self.name = name
+        self.name = name or ''
```

Comment:
With a high-level look at the code, I think it will cause misbehavior in the Asset API. I'll do a deeper review and test on Monday.